python/lsst/ap/association/diaForcedSource.py

```diff
@@ -264,3 +266,31 @@ def _calibrate_and_merge(self,
         output_catalog.drop(columns=self.config.dropColumns, inplace=True)
 
         return output_catalog
+
+    def _trim_to_bbox(self, catalog, exposure):
+        """Remove DiaForcedSources that are outside of the bounding box region.
+
+        Paramters
+        ---------
+        catalog : `pandas.DataFrame`
+            DiaForcedSources to check against the exposure bounding box.
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to check against.
+
+        Returns
+        -------
+        output : `pandas.DataFrame`
+            DataFrame trimmed to only the objects within the exposure bounding
+            box.
+        """
+        bbox = geom.Box2D(exposure.getBBox())
+        minX = bbox.getMinX()
+        minY = bbox.getMinY()
+        maxX = bbox.getMaxX()
+        maxY = bbox.getMaxY()
+
+        xS = catalog.loc[:, "x"]
+        yS = catalog.loc[:, "y"]
+
+        return catalog[np.logical_and(np.logical_and(minX < xS, maxX > xS),
+                                      np.logical_and(minY < yS, maxY > yS))]
```

Comment:
`Box2D` has a vector `contains` method, so I think you can simplify this to something like:

```
return catalog[bbox.contains(xS, yS)]
```

The loop over `xS` and `yS` is [done in C++](https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html#vectorizing-functions), so this should also be faster than the current code.

Reply:
Thanks, didn't see the vectorized bbox.contains.